clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -168,6 +178,44 @@ public void testCommitted_ExceptionThrown() {
         }
     }
 
+    @Test
+    public void testWakeupBeforeCallingPoll() {
+        final TopicPartition tp = new TopicPartition("foo", 3);
+        consumer.assign(singleton(tp));
+
+        consumer.wakeup();
+
+        assertThrows(WakeupException.class, () -> consumer.poll(Duration.ofMinutes(1)));
+    }
+
+    @Test
+    public void testWakeupAfterEmptyFetch() {
```

Comment:
Do we not want to test `wake-up` with a non-empty fetch?

Reply:
I forgot about that unit test! Thanks!